### PR TITLE
CR description of Application Connector

### PR DESCRIPTION
**Pull request by koala7659**
**Description**

Changes proposed in this pull request:

- Update the AC module documentation to the current state of the module

**Related issue(s)**

See also: 
- https://github.com/kyma-project/application-connector-manager/issues/41

### CODE REVIEW

File: docs/user/resources/06-30-application-connector.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# ApplicationConnector CustomResourceDefinition (CRD)
```
Comment (grego952) [inline docs/user/resources/06-30-application-connector.md:1]:
```suggestion
# ApplicationConnector
```
The way it's written in other modules' documentation.

File: docs/user/resources/06-30-application-connector.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+The applicationconnectors.operator.kyma-project.io CustomResourceDefinition (CRD) is a comprehensive specification that defines the structure and format used to manage the configuration and status of the Application Connector Module within your Kyma environment.
```
Comment (grego952) [inline docs/user/resources/06-30-application-connector.md:3]:
```suggestion
The `applicationconnectors.operator.kyma-project.io` CustomResourceDefinition (CRD) is a comprehensive specification that defines the structure and format used to manage the configuration and status of the Application Connector module within your Kyma environment.
```

File: docs/user/resources/06-30-application-connector.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+# Purpose
+
+The CRD serves as the blueprint for creating custom resources (CRs) specific to Application Connectors. These CRs encapsulate the configuration settings and real-time statuses of the Application Connector Module, providing essential insights into the module's behavior on the Kubernetes cluster.
```
Comment (grego952) [inline docs/user/resources/06-30-application-connector.md:7]:
```suggestion
The CRD serves as the blueprint for creating custom resources (CRs) specific to Application Connector. These CRs encapsulate the configuration settings and real-time statuses of the Application Connector module, providing essential insights into the module's behavior on the Kubernetes cluster.
```

File: docs/user/resources/06-30-application-connector.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+```bash
+kubectl get crd applicationconnectors.operator.kyma-project.io -o yaml
+```
+
+## Custom resource specification
+
+| Path | Description | Required |
+| ---- | ----------- | -------- |
+| domainName | The name of the domain that will be used by the virtual service and gateway resources. On gardener cluster this field is autopopulated. | false |
```
Comment (grego952) [inline docs/user/resources/06-30-application-connector.md:19]:
```suggestion
| domainName | The domain name used by the virtual service and gateway resources. On the Gardener cluster, this field is autopopulated. | false |
```

File: docs/user/resources/06-30-application-connector.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+| appConnValidator.logLevel | Determines the level of verbosity or detail for logging messages generated by __central application connector validator__; one of `debug`, `panic`, `fatal`, `error`, `warn`, `info`, `debug` | false |
```
Comment (grego952) [inline docs/user/resources/06-30-application-connector.md:20]:
```suggestion
| appConnValidator.logLevel | Determines the level of verbosity or detail for logging messages generated by the central Application Connectivity Validator; one of `debug`, `panic`, `fatal`, `error`, `warn`, `info`, `debug` | false |
```
Is there a difference between `Application Connectivity Validator` and `Application Connector Validator`? If not, let's stick to one.

File: docs/user/resources/06-30-application-connector.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+| appConnValidator.logFormat | Allows to specify the format in which log messages are generated by the __central application connectivity validator__; one of: `json`, `text`  | false |
```
Comment (grego952) [inline docs/user/resources/06-30-application-connector.md:21]:
```suggestion
| appConnValidator.logFormat | Allows for specifying the format in which the central Application Connectivity Validator generates log messages; one of: `json`, `text`  | false |
```

File: docs/user/resources/06-30-application-connector.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+| appGateway.logLevel | Determines the level of verbosity or detail for logging messages generated by __central application gateway__; one of `debug`, `panic`, `fatal`, `error`, `warn`, `info`, `debug` | false |
```
Comment (grego952) [inline docs/user/resources/06-30-application-connector.md:22]:
```suggestion
| appGateway.logLevel | Determines the level of verbosity or detail for logging messages generated by the central Application Gateway; one of `debug`, `panic`, `fatal`, `error`, `warn`, `info`, `debug` | false |
```

File: docs/user/resources/06-30-application-connector.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+| appGateway.proxyTimeout | Specifies the maximum duration, for which the __central application gateway__ will wait for a response from a external system before timing out | false |
```
Comment (grego952) [inline docs/user/resources/06-30-application-connector.md:23]:
```suggestion
| appGateway.proxyTimeout | Specifies the maximum duration for which the central Application Gateway waits for a response from an external system before timing out | false |
```

File: docs/user/resources/06-30-application-connector.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+| appGateway.requestTimeout | Specifies the maximum duration, for which the __central application gateway__ will wait for a response from a client request before timing out. | false |
```
Comment (grego952) [inline docs/user/resources/06-30-application-connector.md:24]:
```suggestion
| appGateway.requestTimeout | Specifies the maximum duration for which the central Application Gateway waits for a response from a client request before timing out. | false |
```

File: docs/user/resources/06-30-application-connector.md (hunk 5 above)
Comment (grego952) [inline docs/user/resources/06-30-application-connector.md:20]: Do we need the word `central` before each component? What's the difference between the central Application Connectivity Validator and the regular one?

Comment (koala7659) [reply docs/user/resources/06-30-application-connector.md:20]: This is the remains of old architecture where each application (registered system) was handled by its own validator and gateway instances. It was replaced by central components. I agree that we can stop using 'Central' prefix but we must remain consistent across whole our documentation.

Comment (grego952) [reply docs/user/resources/06-30-application-connector.md]: Ok, so let's remove it :)

File: docs/user/resources/06-30-application-connector.md [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+```bash
+kubectl get crd applicationconnectors.operator.kyma-project.io -o yaml
+```
+
+## Custom resource specification
+
+| Path | Description | Required |
+| ---- | ----------- | -------- |
+| domainName | The domain name used by the virtual service and gateway resources. On the Gardener cluster, this field is autopopulated. | false |
+| appConnValidator.logLevel | Determines the level of verbosity or detail for logging messages generated by the central Application Connectivity Validator; one of `debug`, `panic`, `fatal`, `error`, `warn`, `info`, `debug` | false |
```
Comment (grego952) [inline docs/user/resources/06-30-application-connector.md:20]:
```suggestion
| appConnValidator.logLevel | Determines the level of verbosity or detail for logging messages generated by Application Connectivity Validator; one of `debug`, `panic`, `fatal`, `error`, `warn`, `info`, `debug` | false |
```

File: docs/user/resources/06-30-application-connector.md [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+| appConnValidator.logFormat | Allows for specifying the format in which the central Application Connectivity Validator generates log messages; one of: `json`, `text`  | false |
+| appGateway.logLevel | Determines the level of verbosity or detail for logging messages generated by the central Application Gateway; one of `debug`, `panic`, `fatal`, `error`, `warn`, `info`, `debug` | false |
```
Comment (grego952) [inline docs/user/resources/06-30-application-connector.md:22]:
```suggestion
| appGateway.logLevel | Determines the level of verbosity or detail for logging messages generated by Application Gateway; one of `debug`, `panic`, `fatal`, `error`, `warn`, `info`, `debug` | false |
```

File: docs/user/resources/06-30-application-connector.md [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+kubectl get crd applicationconnectors.operator.kyma-project.io -o yaml
+```
+
+## Custom resource specification
+
+| Path | Description | Required |
+| ---- | ----------- | -------- |
+| domainName | The domain name used by the virtual service and gateway resources. On the Gardener cluster, this field is autopopulated. | false |
+| appConnValidator.logLevel | Determines the level of verbosity or detail for logging messages generated by the central Application Connectivity Validator; one of `debug`, `panic`, `fatal`, `error`, `warn`, `info`, `debug` | false |
+| appConnValidator.logFormat | Allows for specifying the format in which the central Application Connectivity Validator generates log messages; one of: `json`, `text`  | false |
```
Comment (grego952) [inline docs/user/resources/06-30-application-connector.md:21]:
```suggestion
| appConnValidator.logFormat | Allows for specifying the format in which Application Connectivity Validator generates log messages; one of: `json`, `text`  | false |
```

File: docs/user/resources/06-30-application-connector.md [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+| appGateway.logLevel | Determines the level of verbosity or detail for logging messages generated by the central Application Gateway; one of `debug`, `panic`, `fatal`, `error`, `warn`, `info`, `debug` | false |
+| appGateway.proxyTimeout | Specifies the maximum duration for which the central Application Gateway waits for a response from an external system before timing out | false |
```
Comment (grego952) [inline docs/user/resources/06-30-application-connector.md:23]:
```suggestion
| appGateway.proxyTimeout | Specifies the maximum duration for which Application Gateway waits for a response from an external system before timing out | false |
```

File: docs/user/resources/06-30-application-connector.md [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+| appGateway.requestTimeout | Specifies the maximum duration for which the central Application Gateway waits for a response from a client request before timing out. | false |
```
Comment (grego952) [inline docs/user/resources/06-30-application-connector.md:24]:
```suggestion
| appGateway.requestTimeout | Specifies the maximum duration for which Application Gateway waits for a response from a client request before timing out. | false |
```

**Review by grego952 — APPROVED** [review]

**Pull request merged**